packages/flutter_tools/lib/runner.dart

```diff
@@ -111,10 +111,23 @@ Future<int> run(
 
           // TODO(eliasyishak): Set the telemetry for the unified_analytics
           //  package as well, the above will be removed once we have
-          //  fully transitioned to using the new package
+          //  fully transitioned to using the new package, https://github.com/flutter/flutter/issues/128251
           await globals.analytics.setTelemetry(true);
         }
 
+        // Ensure that anyone opted out of package:unified_analytics is also
+        // opted out of legacy analytics and that analytics is not being suppressed
+        //
+        // Check that the instance for unified_analytics is not a NoOp instance which
+        // could indicate we are running on ci where legacy analytics events are being
+        // logged to a local log file
+        // TODO(eliasyishak): remove once GA3 sunset, https://github.com/flutter/flutter/issues/128251
+        if (!globals.analytics.telemetryEnabled &&
+            globals.flutterUsage.enabled) {
+          AnalyticsConfigEvent(enabled: false).send();
```

Comment:
I'm guessing this sends the opt out event. Will sending the opt-out event count someone as an active user? If so, I'm guessing that this will not actually immediately reduce our count of monthly active users on GA3. Should we also send a special new event (e.g. `opted_from_ga3_becaused_opted_out_ga4` or some such) that we can use specifically to do the math?

Reply:
Yeah I think that would be a great idea, looking into how to add a new event for GA3 now.

Do you think it would still be worth opting out the GA3 users if we have this then? We could leave them opted in and just do a unique count of users that fall into the case where they are opted out of GA4 but remain opted in GA3 with this new event.

Reply:
> We could leave them opted in and just do a unique count of users that fall into the case where they are opted out of GA4 but remain opted in GA3 with this new event.

Yes, this makes sense to me.

Reply:
Cool, we can also replace the tests by doing so too.  Just waiting for the GA3 export job to confirm the new event works as expected

Reply:
Took a while for the tables to export but i can confirm that this new events can be something we use to get an aggregate count

<img width="404" alt="image" src="https://github.com/flutter/flutter/assets/42216813/730e9324-0cd5-40cb-8ad2-065a5dad9bcd">

I also reverted the test deletions so all this PR is looking to do now is just send a new event